test/fake_clock.h

```diff
@@ -0,0 +1,36 @@
+#ifndef FAKE_CLOCK_H_
+#define FAKE_CLOCK_H_
+
+#include <chrono>
+#include <cstdint>
+#include <ratio>
+
+namespace google {
+namespace testing {
+
+class FakeClock {
+ public:
+  // Requirements for Clock, see:
+  // http://en.cppreference.com/w/cpp/named_req/Clock.
+  using rep = uint64_t;
```

Comment:
This could, in fact, be:
```c++
  using rep = time::seconds::rep;
  using period = time::seconds::period;
  using duration = time::seconds;
```
(you'd need to include `../src/time.h`).